pkg/util/slowset.go

```diff
@@ -25,68 +25,86 @@ const (
 	slowSetSleepDelay = 100 * time.Millisecond
 )
 
-type SlowSet struct {
+type FailureTimestamp[T ~string] struct {
+	FailureType T
+	Timestamp   time.Time
+}
+
+type SlowSet[T ~string] struct {
 	sync.RWMutex
 	// retentionTime is the time after which an item will be removed from the set
 	// this indicates, how long before an operation on pvc can be retried.
 	retentionTime time.Duration
-	workSet       map[string]time.Time
+	workSet       map[string]FailureTimestamp[T]
```

Comment:
Why the slowset has multiple entries per PVC?
Either a PVC is in the slowset (the last expansion was infeasible) or it's not (anything else - success, feasible error, resize has not started yet).